odbc_fdw.c

```diff
@@ -155,9 +182,208 @@ odbc_fdw_handler(PG_FUNCTION_ARGS)
 	fdwroutine->IterateForeignScan = odbcIterateForeignScan;
 	fdwroutine->ReScanForeignScan = odbcReScanForeignScan;
 	fdwroutine->EndForeignScan = odbcEndForeignScan;
+	fdwroutine->ImportForeignSchema = odbcImportForeignSchema;
 	PG_RETURN_POINTER(fdwroutine);
 }
 
+void
+init_odbcFdwOptions(odbcFdwOptions* options)
+{
+	options->dsn          = NULL;
+	options->driver       = NULL;
+	options->host         = NULL;
+	options->port         = NULL;
+	options->database     = NULL;
+	options->username     = NULL;
+	options->password     = NULL;
+	options->schema       = NULL;
+	options->table        = NULL;
+	options->prefix       = NULL;
+	options->sql_query    = NULL;
+	options->sql_count    = NULL;
+	options->mapping_list = NIL;
+}
+
+void
+copy_odbcFdwOptions(odbcFdwOptions* to, odbcFdwOptions* from)
+{
+	if (to && from)
+	{
+		to->dsn          = from->dsn;
+		to->driver       = from->driver;
+		to->host         = from->host;
+		to->port         = from->port;
+		to->database     = from->database;
+		to->username     = from->username;
+		to->password     = from->password;
+		to->schema       = from->schema;
+		to->table        = from->table;
+		to->prefix       = from->prefix;
+		to->sql_query    = from->sql_query;
+		to->sql_count    = from->sql_count;
+		to->mapping_list = from->mapping_list;
+	}
```

Comment:
In this case we're just copying the pointers and letting PG dispose the memory in block (memory context-wise). But I have yet to review memory handling, specially for IMPORT FOREIGN SCHEMA to make sure to memory leaks exist.